src/AdoNetCore.AseClient/AseErrorCollection.cs

```diff
@@ -53,5 +53,22 @@ public IEnumerator GetEnumerator()
                 return _errors[index];
             }
         }
+
+        internal AseError GetMostSevereError()
```

Comment:
Refer to comment on `AseException`, you could do this check upon construction of `AseErrorCollection` and store the result (or the resulting index).